tests/test_statistics.F90

```diff
@@ -0,0 +1,70 @@
+! (C) Copyright 2022- ECMWF.
+! (C) Copyright 2022- Meteo-France.
+! (C) Copyright 2023- NVIDIA
+!
+! This software is licensed under the terms of the Apache Licence Version 2.0
+! which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+! In applying this licence, ECMWF does not waive the privileges and immunities
+! granted to it by virtue of its status as an intergovernmental organisation
+! nor does it submit to any jurisdiction.
+
+PROGRAM TEST_STATISTICS
+
+USE FIELD_ABORT_MODULE
+USE FIELD_STATISTICS_MODULE
+USE FIELD_MODULE
+USE PARKIND1
+USE FIELD_FACTORY_MODULE
+USE FIELD_ACCESS_MODULE
+
+IMPLICIT NONE
+CLASS(FIELD_4IM), POINTER :: YLF1 => NULL(), YLF2 => NULL ()
+INTEGER (KIND=JPIM) :: UBOUNDS (4) = [2,10,10,10]
+INTEGER (KIND=JPIM) :: IDATA0 (5,3,2,9)
+INTEGER (KIND=JPIM), POINTER :: IDATA (:,:,:,:)
+
+FIELD_STATISTICS_ENABLE = .TRUE.
+
+CALL FIELD_NEW (YLF1, UBOUNDS=UBOUNDS, PERSISTENT=.TRUE.)
+
+CALL ASSERT (FIELD_STATISTICS%NFIELDS_CUR==1)
+
+CALL FIELD_NEW (YLF2, DATA=IDATA0, PERSISTENT=.TRUE.)
+
+CALL ASSERT (FIELD_STATISTICS%NFIELDS_CUR==2)
+
+CALL ASSERT (FIELD_STATISTICS%MEMORY_DEVICE_CUR==0)
+
+IDATA => GET_DEVICE_DATA_RDONLY (YLF1)
+
+CALL ASSERT (FIELD_STATISTICS%MEMORY_DEVICE_CUR==INT(KIND(IDATA),JPIB)*SIZE(IDATA,KIND=JPIB))
+
+IDATA => GET_DEVICE_DATA_RDONLY (YLF2)
+
+CALL ASSERT (FIELD_STATISTICS%MEMORY_DEVICE_CUR==INT(JPIM,JPIB)*PRODUCT(INT(UBOUNDS,JPIB))+INT(JPIM,JPIB)*SIZE(IDATA0,KIND=JPIB))
+
+CALL FIELD_DELETE (YLF1)
+
+CALL ASSERT (FIELD_STATISTICS%NFIELDS_CUR==1)
+
+CALL FIELD_DELETE (YLF2)
+
+CALL ASSERT (FIELD_STATISTICS%NFIELDS_CUR==0)
+
+CALL ASSERT (FIELD_STATISTICS%NFIELDS_MAX==2)
+
+CALL ASSERT (FIELD_STATISTICS%MEMORY_DEVICE_CUR==0)
+
+CALL ASSERT (FIELD_STATISTICS%MEMORY_DEVICE_MAX==PRODUCT(INT(UBOUNDS,JPIB))*INT(JPIM,JPIB)+SIZE(IDATA0,KIND=JPIB)*INT(JPIM,JPIB))
+
+CONTAINS 
+
+SUBROUTINE ASSERT (LDCOND)
```

Comment:
[Optional] This makes test much nicer to read. Could we maybe move this to FIELD_ABORT_MODULE or similar to make this widely available in the test base? Just a thought...